packages/idle-timer/src/mekeIdleTimer.ts

```diff
@@ -0,0 +1,156 @@
+ *    @param params.element: HTMLElement; DOM element to which the event listeners will be attached. It defaults to document.
+ *    @param params.startsManually: boolean; requires the event-listeners to be bound manually by using the start method (see @returns), instead of on mount. It defaults to false.
+ * }
+ * @returns - returns an object with several methods and accessors
+ * {
+ *    @returns.isIdle: Accessor<boolean>; shows when the user is idle
+ *    @returns.isPrompted: Accessor<boolean>; tells whether params.onPrompt has been called or not, and consequently if the promptTimeout countdown has started
+ *    @returns.reset: () => void; resets timers, doesn't trigger onActive.
+ *    @returns.start: () => void; adds listeners and starts timers, doesn't trigger onActive.
+ *    @returns.stop: () => void; removes listeners and cleans up the timers, doesn't trigger onActive.
+ * }
+ */
+export const makeIdleTimer = ({
+  element,
+  events = EVENTS,
+  idleTimeout = FIFTEEN_MINUTES,
+  promptTimeout = 0,
+  onActive,
+  onIdle,
+  onPrompt,
+  startManually = false,
+}: IdleTimerOptions): IdleTimerReturn => {
+  const [listenersAreOn, setListenersAreOn] = createSignal(false);
+  const [isPrompted, setIsPrompted] = createSignal(false);
+  const [isIdle, setIsIdle] = createSignal(false);
+
+  let idle: ReturnType<typeof setTimeout>;
+  let prompt: ReturnType<typeof setTimeout>;
+  let lastThrottle: number = 0;
+
+  function shouldPreventRunning() {
+    const now = new Date().getTime();
+    const shouldPrevent = now - lastThrottle < THROTTLE_DELAY;
+    if (!shouldPrevent) lastThrottle = now;
+
+    return shouldPrevent;
+  }
+
+  function handleEvent(evt: Event) {
+    if (shouldPreventRunning()) return;
+    if (isIdle()) onActive?.(evt);
+    if (!isPrompted()) timerReset(evt);
+  }
+
+  function addListeners() {
+    if (listenersAreOn()) return;
+    const target = element ?? document;
+
+    for (const evt of events) {
+      target.addEventListener(evt, handleEvent);
+    }
+
+    setListenersAreOn(true);
+  }
+
+  function timerReset(evt: Event) {
+    if (!listenersAreOn()) return;
+
+    timerCleanup();
+    cleanState();
+    setIdleTimer(evt);
+  }
+
+  function timerCleanup() {
+    if (typeof idle === 'number') clearTimeout(idle);
+    if (typeof prompt === 'number') clearTimeout(prompt);
+  }
+
+  function setIdleTimer(evt: Event) {
+    idle = setTimeout(() => {
+      setIsPrompted(true)
+      onPrompt?.(evt)
+      setPromptTimer(evt)
+    }, idleTimeout);
+  }
+
+  function setPromptTimer(evt: Event) {
+    prompt = setTimeout(() => {
+      setIsIdle(true);
+      setIsPrompted(false)
+      onIdle?.(evt);
+    }, promptTimeout)
+  }
+
+  function cleanState() {
+    if (isIdle()) setIsIdle(false);
+    if (isPrompted()) setIsPrompted(false);
```

Comment:
Oh that's a very good point, thx.
I changed as requested, sometimes subscription can be tricky ^^
Ok I was wondering 

> if (isIdle()) check is unnecessary since signals already do that internally—assigning the same variable won't cause an update.
I didn't know that, that is helpful for the future, thx